app/javascript/articles/components/SaveButton.jsx

```diff
@@ -8,20 +8,23 @@ export class SaveButton extends Component {
 
     const { isBookmarked } = props;
 
-    this.state = { buttonText: isBookmarked ? 'Saved' : 'Save' };
+    this.state = {
+      buttonText: isBookmarked ? 'Saved' : 'Save',
+      hovering: false,
```

Comment:
The `hovering` state property is not necessary. We can leverage native browser events to handle this, i.e. `mouseMove`